lib/nxt_state_machine/state_machine.rb

```diff
@@ -0,0 +1,197 @@
+module NxtStateMachine
+  class StateMachine
+    def initialize(name, class_context, event_registry, **opts)
+      @name = name
+      @class_context = class_context
+      @options = opts
+
+      @states = NxtStateMachine::StateRegistry.new
+      @transitions = []
+      @events = event_registry
+      @callbacks = CallbackRegistry.new
+      @error_callback_registry = ErrorCallbackRegistry.new
+      @defuse_registry = DefuseRegistry.new
+
+      @initial_state = nil
+    end
+
+    attr_reader :class_context, :events, :options, :callbacks, :name, :error_callback_registry, :defuse_registry
+    attr_accessor :initial_state
+
+    def get_state_with(method = nil, &block)
```

Comment:
I guess it would be nice to have comments above such methods, as it's done in Rails source code for example.
This is not business-related code, but rather metaprogramming-dense code, so it might be useful for developers, as well as creators of this code, to understand better what each method does (and maybe even what it returns).